asab/metrics/prometheus.py

```diff
@@ -0,0 +1,111 @@
+import asab
+import re
+import logging
+
+
+L = logging.getLogger(__name__)
+
+
+def validate_format(name):
+	regex = r"[a-zA-Z:][a-zA-Z0-9_:]*"
+	match = re.fullmatch(regex, name)
+	if match is None:
+		L.warning("Invalid Prometheus format. {} must match the regex [a-zA-Z:][a-zA-Z0-9_:]*".format(name))
+		regex_sub = r"[^a-zA-Z0-9_:]"
+		name = re.sub(regex_sub, "_", name)
+		name = name.lstrip("_0123456789")
+		if name.endswith(("total", "created")):
+			L.warning("Invalid OpenMetrics format. Name MUST NOT end with total or created.")
+	return name
+
+
+def validate_value(value):
+	return isinstance(value, [int, float])
+
+
+def get_labels(tags):
+	labels_str = "{"
+	for tag in tags.keys():
+		if tag in {"host", "unit", "help"}:
+			continue
+		else:
+			label_name = validate_format(tag)
+			label_value = validate_format(tags.get(tag))
+			labels_str += '{}="{}",'.format(label_name, label_value)
+	labels_str = labels_str.rstrip(",")
+	labels_str += "}"
+	if len(labels_str) <= 2:
+		return None
+	else:
+		return labels_str
+
+
+def metric_to_text(metric, type):
+	metric_lines = []
+	m_name = metric.get("Name")
+	tags = metric.get("Tags")
+	labels_str = get_labels(tags)
+
+	for v_name, value in metric.get("Values").items():
+		if validate_value is False:
+			L.warning("Invalid OpenMetrics format. Value must be float or integer. {} omitted.".format(m_name))
+			continue
+		else:
+			name = "_".join([m_name, v_name])
+			name = validate_format(name)
+			# If a unit is specified it MUST be provided in a UNIT metadata line. In addition, an underscore and the unit MUST be the suffix of the MetricFamily name.
+			if tags.get("unit"):
```

Comment:
This condition in both cases do `metric_lines.append(..`. It is possible to separate this to another function and provide test for that. Then you could replace condition with:
`metric_lines.append(your_function(unit, type))`